src/Raven.Server/ServerWide/Commands/PeriodicBackup/UpdatePeriodicBackupStatusCommand.cs

```diff
@@ -29,9 +64,172 @@ protected override BlittableJsonReaderObject GetUpdatedValue(long index, RawData
             return context.ReadObject(PeriodicBackupStatus.ToJson(), GetItemId());
         }
 
+        public override object GetState()
+        {
+            return _backupDetailsIdsToDelete;
+        }
+
+        public override unsafe void Execute(ClusterOperationContext context, Table items, long index, RawDatabaseRecord record, RachisState state, out object result)
+        {
+            base.Execute(context, items, index, record, state, out result);
+
+            CurrentAndTemporarySavedEntries.ForEach(
+                backupHistoryEntryToAdd =>
+                {
+                    var itemKey = backupHistoryEntryToAdd.GenerateItemKey();
```

Comment:
IMO, the way this is persisted, makes in complex and a lot of data can be stored in one document, imagine a case when I have a full backup once a month and an incremental every 15min?

Each of those should be stored separately, yet should be identified by a key quicker for easy access, paging, removal, etc